docs/how_to/keys/grok.md

```diff
@@ -0,0 +1,106 @@
+# How to get and use a Grok.ai API Key
```

Comment:
I wonder if we should rename this to:
"How to get and use an xAI Grok API key" so we're clearly calling it xAI since that's the company?